### PR TITLE
Duplicate views: Use Calypso admin menu for untangled WP-Admin screens

### PR DESCRIPTION
**Pull request by BogdanUngureanu**
Integrate the navigation experiment for Simple and WoA sites - users part of the treatment will now receive the Core WP-Admin interface but with the Calypso admin menu.

## Proposed changes:

* Enable wpcom_admin_interface option for screens in an allowlist
* Keep the Calypso navigation for them

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Use the Jetpack downloader instructions to test the changes on Simple and Atomic sites
* Enable the experiment treatment on your user: 22124-explat-experiment
* Go to a Simple Site and make sure it's using Calypso for Posts
* Sandbox it and make sure that the admin menu is still Calypso
* Click on Posts and make sure that Calypso admin menu is used but the page is WP-Admin
* Make sure that the untangling changes are applied on the page (e.g. there's no Classic Editor in the post quick links)
* Repeat the steps for an Atomic site
* Since we don't have untangling changes there, just make sure that /edit.php loads and we still use Calypso admin menu

### CODE REVIEW

File: projects/packages/masterbar/src/class-main.php [hunk 1]
```diff
@@ -29,7 +29,10 @@ public static function init() {
 
 		new Admin_Color_Schemes();
 
-		if ( get_option( 'wpcom_admin_interface' ) === 'wp-admin' ) {
+		remove_filter( 'pre_option_wpcom_admin_interface', 'wpcom_admin_interface_pre_get_option' );
+		$is_wp_admin_interface = get_option( 'wpcom_admin_interface' ) === 'wp-admin';
+		add_filter( 'pre_option_wpcom_admin_interface', 'wpcom_admin_interface_pre_get_option', 10 );
```
Comment (mmtr) [inline projects/packages/masterbar/src/class-main.php]: `wpcom_admin_interface_pre_get_option` doesn't exist on Jetpack sites, so I wonder we should add another `function_exists` check here.

Not sure if this file is loaded on Jetpack sites (maybe by the `admin-menu` endpoint?) though.

Comment (BogdanUngureanu) [reply projects/packages/masterbar/src/class-main.php]: Added a check just in case.

**Comment by mmtr** [comment]
I'm getting an error on Atomic sites when applying the changes to wpcomsh:

```
[06-Dec-2024 16:31:44 UTC] PHP Fatal error:  Uncaught Error: Call to undefined function wpcom_is_using_default_admin_menu() in /srv/htdocs/wp-content/mu-plugins/wpcomsh-dev/feature-plugins/masterbar.php:96
Stack trace:
#0 /srv/htdocs/wp-content/mu-plugins/wpcomsh-dev/vendor/automattic/wc-calypso-bridge/class-wc-calypso-bridge-shared.php(53): wpcomsh_activate_nav_unification()
#1 /srv/htdocs/wp-content/mu-plugins/wpcomsh-dev/vendor/automattic/wc-calypso-bridge/class-wc-calypso-bridge-shared.php(30): WC_Calypso_Bridge_Shared->__construct()
#2 /srv/htdocs/wp-content/mu-plugins/wpcomsh-dev/vendor/automattic/wc-calypso-bridge/class-wc-calypso-bridge-shared.php(221): WC_Calypso_Bridge_Shared::instance()
#3 /srv/htdocs/wp-content/mu-plugins/wpcomsh-dev/vendor/automattic/wc-calypso-bridge/class-wc-calypso-bridge.php(129): require_once('/srv/htdocs/wp-...')
#4 /srv/htdocs/wp-content/mu-plugins/wpcomsh-dev/vendor/automattic/wc-calypso-bridge/class-wc-calypso-bridge.php(104): WC_Calypso_Bridge->includes()
#5 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(324): WC_Calypso_Bridge->initialize('')
#6 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#7 /srv/htdocs/__wp__/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#8 /srv/htdocs/__wp__/wp-settings.php(559): do_action('plugins_loaded')
#9 /srv/htdocs/wp-config.php(87): require_once('/srv/htdocs/__w...')
#10 /srv/htdocs/__wp__/wp-load.php(55): require_once('/srv/htdocs/wp-...')
#11 /srv/htdocs/__wp__/wp-admin/admin.php(34): require_once('/srv/htdocs/__w...')
#12 {main}
```

Before applying the changes to wpcomsh, I applied them to Jetpack and jetpack-mu-wpcom to make sure that the `wpcom_is_using_default_admin_menu` was defined, but somehow wpcomsh cannot locate it

**Comment by mmtr** [comment]
I get a different error now on Atomic:

```
PHP Fatal error:  Uncaught TypeError: call_user_func_array(): Argument #1 ($callback) must be a valid callback, function "wpcom_admin_interface_pre_get_option" not found or invalid function name in /srv/htdocs/__wp__/wp-includes/class-wp-hook.php:326
Stack trace:
#0 /srv/htdocs/__wp__/wp-includes/plugin.php(205): WP_Hook->apply_filters('calypso', Array)
#1 /srv/htdocs/__wp__/wp-includes/option.php(132): apply_filters('pre_option_wpco...', false, 'wpcom_admin_int...', false)
#2 /srv/htdocs/wp-content/mu-plugins/wpcomsh-dev/feature-plugins/class-theme-homepage-switcher.php(44): get_option('wpcom_admin_int...')
#3 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(324): Theme_Homepage_Switcher->register_hooks('')
#4 /srv/htdocs/__wp__/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#5 /srv/htdocs/__wp__/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#6 /srv/htdocs/__wp__/wp-settings.php(559): do_action('plugins_loaded')
#7 /srv/htdocs/wp-config.php(87): require_once('/srv/htdocs/__w...')
#8 /srv/htdocs/__wp__/wp-load.php(55): require_once('/srv/htdocs/wp-...')
#9 /srv/htdocs/__wp__/wp-admin/admin.php(34): require_once('/srv/htdocs/__w...')
#10 {main}
  thrown in /srv/htdocs/__wp__/wp-includes/class-wp-hook.php on line 326
```

**Comment by mmtr** [comment]
Seems that the errors were specific to my old dev Atomic sites – cannot replicate on a new site.

**Review by mmtr — DISMISSED** [review]
This tests well for me.

This stops pointing the "Add a new post" link to the iframed editor, which is unwanted, but can be fixed in a follow-up.

**Pull request merged**